src/NATS.Client.Core/NatsRequestManyAsyncExtensions.cs

```diff
@@ -0,0 +1,143 @@
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using NATS.Client.Core.Internal;
+
+namespace NATS.Client.Core;
+
+public static class NatsRequestManyAsyncExtensions
+{
+    // RequestManyAsync methods
+    // Same as PublishAsync with the following changes
+    // - Response is IAsyncEnumerable<NatsMsg>
+    // - PubOpts is called requestOpts
+    // - add SubOpts replyOpts
+    //   - if replyOpts.Timeout == null then set to NatsOptions.RequestTimeout
+    public static async IAsyncEnumerable<NatsMsg> RequestManyAsync(
+        this NatsConnection nats,
+        string subject,
+        ReadOnlySequence<byte> payload = default,
+        NatsPubOpts? requestOpts = default, // Not used
```

Comment:
I think this could be used - Headers and Queue Group could be set here right?

Reply:
Yes, spot on! Thank you. I forgot about headers when publishing. Queue group doesn't apply to pub-opts. But a good question. Does it make sense to have a queue group on an inbox subscription? If it does, how should we go about it when our inbox is mux-ed and shared, as in this case?

Reply:
Oops - yes - queue group is on subscribe.  Good catch.  No, I don't think an Inbox Subscription really needs to support a Queue Group, but it's also not a big deal if it does